packages/azure-node/src/generators/app-insights/generator.ts

```diff
@@ -77,6 +77,12 @@ export default async function appInsightsGenerator(
 
     const mainFunction = customServer.getFunction('main');
 
+    if (appInsightsKey.length === 0) {
```

Comment:
The schema supports the following:
```json
{
  "pattern": "\\d+",
  "minLength": 10,
  "maxLength": 100
}
```

Might be better easier to use this an enforce capitals and underscores only? Thoughts?

Reply:
I'll add this into the schema, it's much cleaner if the schema supports some form of error handling for scenarios like this